extensions/shared/upgrade-nudge/index.jsx

```diff
@@ -81,13 +81,27 @@ export default compose( [
 		// The editor for CPTs has an `edit/` route fragment prefixed
 		const postTypeEditorRoutePrefix = [ 'page', 'post' ].includes( postType ) ? '' : 'edit';
 
+		const isWpcom = get( window, [ '_currentSiteType' ] ) === 'simple';
+
+		// Post-checkout: redirect back here
+		const redirect_to = isWpcom
+			? '/' +
+			  compact( [ postTypeEditorRoutePrefix, postType, getSiteFragment(), postId ] ).join( '/' )
+			: addQueryArgs(
+					window.location.protocol +
```

Comment:
Devil's advocate: Why even bother constructing the URL from 'reliable' data and not just `window.location.href` for the entire `redirect_to` URL I'm building?